src/Components/Aspire.StackExchange.Redis/TempOpenTelemetry.cs

```diff
@@ -0,0 +1,50 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.CompilerServices;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Options;
+using OpenTelemetry.Instrumentation.StackExchangeRedis;
+using OpenTelemetry.Trace;
+using StackExchange.Redis;
+
+namespace Aspire.StackExchange.Redis;
+
+// This is a temporary solution to add instrumentation for keyed service.
+// This class needs to get removed once this issue got solved and published.
+// https://github.com/open-telemetry/opentelemetry-dotnet-contrib/issues/1451
+
+// This code has been copied from here
+//https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/c3a4f335f5a1f4fc784d44cc99b5bf44e872a66c/src/OpenTelemetry.Instrumentation.StackExchangeRedis/TracerProviderBuilderExtensions.cs#L109
+internal static class TempOpenTelemetry
+{
+    [UnsafeAccessor(UnsafeAccessorKind.Constructor)]
+    static extern StackExchangeRedisInstrumentation CreateInstrumentation(IOptionsMonitor<StackExchangeRedisInstrumentationOptions> options);
+
+    public static TracerProviderBuilder AddRedisInstrumentationWithKeyedService(this TracerProviderBuilder builder, object serviceKey)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+
+        builder.ConfigureServices(services =>
+        {
+            services.TryAddSingleton(sp => CreateInstrumentation(sp.GetRequiredService<IOptionsMonitor<StackExchangeRedisInstrumentationOptions>>()));
+        });
+
+        return builder
+            .AddSource("OpenTelemetry.Instrumentation.StackExchangeRedis")
+            .AddInstrumentation(sp =>
+            {
+                var instrumentation = sp.GetRequiredService<StackExchangeRedisInstrumentation>();
+
+                var connection = sp.GetKeyedService<IConnectionMultiplexer>(serviceKey);
+
+                if (connection != null)
+                {
+                    instrumentation.AddConnection(Options.DefaultName, connection);
```

Comment:
```suggestion
                    instrumentation.AddConnection(serviceKey.ToString(), connection);
```

? What happens if there are more than 1 keyed Redis connections?

Reply:
The current implementation for non-keyed redis connection uses Options.DefaultName I wanted to keep the implementation consistence with it.

I think the `name` is used to get named options for `StackExchangeRedisInstrumentationOptions` 
https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/fbfd66495359a88738bc0f5611bc7728c568ce5a/src/OpenTelemetry.Instrumentation.StackExchangeRedis/StackExchangeRedisInstrumentation.cs#L44

and for the `Thread` name responsible for creating redis `Activity`.
https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/fbfd66495359a88738bc0f5611bc7728c568ce5a/src/OpenTelemetry.Instrumentation.StackExchangeRedis/StackExchangeRedisConnectionInstrumentation.cs#L56

> ? What happens if there are more than 1 keyed Redis connections?
The same thing that if there are more than 1 non-keyed redis connections. I don't see any problem. Am I missing something?

Reply:
> I think the name is used to get named options

This is the part I was missing.

I think this fine for now, but when we take https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1457, we should probably be specifying the name, so someone can configure these options separately from the other Redis connections.